docs/topics/jvm/java-to-kotlin-interop.md

```diff
@@ -335,20 +336,17 @@ public class BB8 implements Robot {
 
 ### Compatibility mode for default methods
 
-If there are clients that use your Kotlin interfaces compiled without the new `-Xjvm-default=all` option, then they can
-be incompatible with the same code compiled with this option. 
-
-To avoid breaking the compatibility with such clients, compile your Kotlin code in the _compatibility mode_ by specifying
-the `-Xjvm-default=all-compatibility` compiler option. In this case, all the code that uses the previous version will 
- work fine with the new one. However, the compatibility mode adds some overhead to the resulting bytecode size.
+If there are clients that use your Kotlin interfaces compiled without the `-Xjvm-default=all` option, then they can
+be incompatible with some code compiled with this option. To avoid breaking the compatibility with such clients, 
+use the `-Xjvm-default=all` mode and mark interfaces with the `@JvmDefaultWithCompatibility` annotation. 
+This way you'll be able to add this annotation to all interfaces in the public API once, and you won't need to use any annotations for a new non-public code.
 
-There is no need to consider compatibility for new interfaces, as no clients have used them before.
-You can minimize the compatibility overhead by excluding these interfaces from the compatibility mode.
-To do this, annotate them with the `@JvmDefaultWithoutCompatibility` annotation. Such interfaces compile the same way as 
-with `-Xjvm-default=all`.
+> From Kotlin 1.6.20, you can compile modules in the default mode (the `-Xjvm-default=disable` compiler option) against 
+> modules compiled with the `-Xjvm-default=all` or `-Xjvm-default=all-compatibility` modes.
+>
+{type="note"}
 
-Additionally, in the `all-compatibility` mode you can use `@JvmDefaultWithoutCompatibility` to annotate all interfaces
-which are not exposed in the public API and therefore aren’t used by the existing clients.
+Learn more about compatibility modes in the [`-Xjvm-default` compiler argument's description](https://github.com/JetBrains/kotlin/blob/1.6.20/compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/arguments/K2JVMCompilerArguments.kt).
```

Comment:
Will such link to the conctere line be ok? https://github.com/JetBrains/kotlin/blob/92a8ebc76094197b778738aa95701b65b1d69fcd/compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/arguments/K2JVMCompilerArguments.kt#L309

I was afraid that it might become irrelevant one day.

Reply:
No, it's not ok. A string in a source file of a compiler is not user's documentation.
This page is user's documentation on the `-Xjvm-default` topic. Why don't you put what you need to explain here?